packages/factory_reset_tools/lib/dbus/drive_manager.dart

```diff
@@ -0,0 +1,203 @@
+import 'dart:async';
+
+import 'package:dbus/dbus.dart';
+import 'package:factory_reset_tools/dbus/drive_data.dart';
+import 'package:factory_reset_tools/dbus/reset_media.dart';
+
+const int minimumRequiredDiskSize = 12 << 30;
```

Comment:
This is 12GiB, but in the loop further down the comment says:
```
// 2. has enough Size (we assume 16G)
```
Is one of those a typo?

Reply:
Probably a typo, right @medicalwei?

Reply:
It was a consideration that disks might be actually smaller than 16GiB or spaces taken by partition metadata. (eg. being 16,000,000,000 bytes instead of 17,179,869,184 bytes.)

But it's fine that you can reimplement that part.

Reply:
It is also worthy to mention that, the mockup does not hide the disks that does not meet the size requirement, they are just greyed out.

Reply:
Is 12 enough? Then I think we should keep that

Reply:
For now yes, as the disk partition we reserve for factory reset is 12GiB

Reply:
So even 4GB should be enough according to @fourdollars, do you think it would be fine if I lowered it that low @medicalwei?

Reply:
IMO @fourdollars says the minimum size should be according to the actual used size of the reset partition, and the number he proposed is just for example.

Reply:
Aah I see, do we know the approximate size of the reset partition?